app/controllers/users_controller.rb

```diff
@@ -38,6 +38,21 @@ def update
     end
   end
 
+  def update_language_settings
+    @user = current_user
+    @tab_list = @user.settings_tab_list
+    @tab = "misc"
+    authorize @user
+    @user.language_settings["preferred_languages"] = params[:user][:preferred_languages].to_a
```

Comment:
I'm concerned with validating language settings when users update their settings.
User can possibly send values that are not valid language codes, so I would like to limit the values.
But if I limit it to the current list (`User::LANGUAGES`), there will be an inconsistency: other language codes can be written to the database when estimating user language on registration, so we would want to keep them at least.

(I'm also not a huge fan of storing languages in the `User::LANGUAGES` constant)